service/controller/resource/azuremachineconditions/conditionvmssready.go

```diff
@@ -0,0 +1,153 @@
+package azuremachineconditions
+
+import (
+	"context"
+
+	"github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2019-07-01/compute"
+	azureconditions "github.com/giantswarm/apiextensions/v3/pkg/conditions/azure"
+	"github.com/giantswarm/microerror"
+	capz "sigs.k8s.io/cluster-api-provider-azure/api/v1alpha3"
+	capi "sigs.k8s.io/cluster-api/api/v1alpha3"
+	capiconditions "sigs.k8s.io/cluster-api/util/conditions"
+
+	"github.com/giantswarm/azure-operator/v5/service/controller/key"
+)
+
+const (
+	VMSSNotFoundReason                 = "VMSSNotFound"
+	VMSSProvisioningStatePrefix        = "VMSSProvisioningState"
+	VMSSProvisioningStateUnknownReason = "VMSSProvisioningStateUnknown"
+	VmssProvisioningStateSucceeded     = string(compute.ProvisioningStateSucceeded)
+	VmssProvisioningStateFailed        = string(compute.ProvisioningStateFailed)
+)
+
+func (r *Resource) ensureVmssReadyCondition(ctx context.Context, cr *capz.AzureMachine) error {
+	r.logDebug(ctx, "ensuring condition %s", azureconditions.VMSSReadyCondition)
+
+	deploymentsClient, err := r.azureClientsFactory.GetDeploymentsClient(ctx, cr.ObjectMeta)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	// Now let's first check ARM deployment state
+	deploymentName := key.MastersVmssDeploymentName
+	isDeploymentSuccessful, err := r.deploymentChecker.CheckIfDeploymentIsSuccessful(ctx, deploymentsClient, cr, deploymentName, azureconditions.VMSSReadyCondition)
+	if err != nil {
+		return microerror.Mask(err)
+	} else if !isDeploymentSuccessful {
+		// Function checkIfDeploymentIsSuccessful that is called above, if it
+		// sees that the deployment is not succeeded, for whatever reason, it
+		// will also set appropriate condition value, so our job here is done.
+		return nil
+	}
+
+	// Deployment is successful, we proceed with checking the actual Azure
+	// VMSS.
+	vmssClient, err := r.azureClientsFactory.GetVirtualMachineScaleSetsClient(ctx, cr.ObjectMeta)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	resourceGroupName := key.ClusterName(cr)
+	vmssName := key.MasterVMSSNameFromClusterAPIObject(cr)
+
+	vmss, err := vmssClient.Get(ctx, resourceGroupName, vmssName)
+	if IsNotFound(err) {
+		r.setVMSSNotFound(ctx, cr, vmssName, azureconditions.VMSSReadyCondition)
+		return nil
+	} else if err != nil {
+		return microerror.Mask(err)
+	}
+
+	// Note: Here we are only checking the provisioning state of VMSS. Ideally
```

Comment:
```suggestion
	// NOTE: Here we are only checking the provisioning state of VMSS. Ideally
```

I think this way it'll highlight in editors by default (at least Vim) :wink: